app/src/main/java/de/xikolo/network/jobs/GetAnnouncementJob.kt

```diff
@@ -0,0 +1,34 @@
+package de.xikolo.network.jobs
+
+import android.util.Log
+import de.xikolo.config.Config
+import de.xikolo.network.ApiService
+import de.xikolo.network.jobs.base.NetworkJob
+import de.xikolo.network.jobs.base.NetworkStateLiveData
+import de.xikolo.network.sync.Sync
+import ru.gildor.coroutines.retrofit.awaitResponse
+
+class GetAnnouncementJob(private val announcementId: String, networkState: NetworkStateLiveData, userRequest: Boolean) : NetworkJob(networkState, userRequest) {
+
+    companion object {
+        val TAG: String = GetAnnouncementJob::class.java.simpleName
+    }
+
+    override suspend fun onRun() {
+        val response = ApiService.instance.getAnnouncement(announcementId).awaitResponse()
```

Comment:
Should this only be executed when the user is logged in? Actually not, but what about a logged-out user trying to load a course announcement?